tests/unit/test_settings.py

```diff
@@ -0,0 +1,345 @@
+from __future__ import annotations
+
+import unittest
+from pathlib import Path
+from typing import Any
+
+import Settings
+from Utils import local_path
+
+
+class TestSettings(unittest.TestCase):
+    def test_settings_init(self):
+        test_settings = Settings.Settings({})
+        self.assertEqual(
+            test_settings.settings_string, test_settings.get_settings_string()
+        )
+        # How to test self.distribution: Distribution = Distribution(self)
+        self.assertFalse(test_settings.numeric_seed is None)
+        self.assertFalse(test_settings.custom_seed)
+
+    def test_old_compress_setting_patch(self):
+        test_settings = Settings.Settings({"compress_rom": "Patch"})
+
+        self.assertTrue(test_settings.create_patch_file)
+        self.assertFalse(test_settings.create_compressed_rom)
+        self.assertFalse(test_settings.create_uncompressed_rom)
+        self.assertRaises(
+            AttributeError,
+            lambda: "compress_rom" in test_settings.__getattribute__("compress_rom"),
+        )
+
+    def test_old_compress_setting_compressed(self):
+        test_settings = Settings.Settings({"compress_rom": "True"})
+
+        self.assertFalse(test_settings.create_patch_file)
+        self.assertTrue(test_settings.create_compressed_rom)
+        self.assertFalse(test_settings.create_uncompressed_rom)
+        self.assertRaises(
+            AttributeError,
+            lambda: "compress_rom" in test_settings.__getattribute__("compress_rom"),
+        )
+
+    def test_old_compress_setting_uncompressed(self):
+        test_settings = Settings.Settings({"compress_rom": "False"})
+
+        self.assertFalse(test_settings.create_patch_file)
+        self.assertFalse(test_settings.create_compressed_rom)
+        self.assertTrue(test_settings.create_uncompressed_rom)
+        self.assertRaises(
+            AttributeError,
+            lambda: "compress_rom" in test_settings.__getattribute__("compress_rom"),
+        )
+
+    def test_default_settings_strict(self):
+        try:
+            Settings.Settings({}, True)
+        except (TypeError, ValueError, BaseException) as e:
```

Comment:
This should probably be `Exception` so it doesn't eat keyboard interrupts.

```suggestion
        except (TypeError, ValueError, Exception) as e:
```